### DISCUSSION TITLE
Expose useIntersection hook for public use

### DISCUSSION THREAD

**Discussion by dohomi** [vercel/next.js#19205] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

The `<Image />` and `<Link />` component make use of `useIntersection`. It would be great if that hook would be publicly available

## Describe the solution you'd like

Expose hook `useIntersection` for public use:
```
import {useIntersection} from 'next/client'
```

**Comment by todortotev** [comment]
Great idea! If accepted I could push a PR/RFC for it?
